stl/inc/functional

```diff
@@ -1207,6 +1207,694 @@ _NODISCARD bool operator!=(nullptr_t, const function<_Fty>& _Other) noexcept {
+        _Checked_move(_Other._Data, _Tmp);
+    }
+
+    _NODISCARD bool _Is_null() const noexcept {
+        return _Data._Impl == &_Null_move_only_function;
+    }
+
+    template <class _Vt>
+    static constexpr bool _Large_function_engaged =
+        alignof(_Vt) > alignof(max_align_t)
+        || sizeof(_Vt) > _Move_only_function_data::_Buf_size<_Vt> || !is_nothrow_move_constructible_v<_Vt>;
+
+    _NODISCARD auto _Get_invoke() const noexcept {
+        return static_cast<const _Impl_t*>(_Data._Impl)->_Invoke;
+    }
+
+    template <class _Vt, class _VtInvQuals>
+    _NODISCARD static constexpr _Impl_t _Create_impl() noexcept {
+        _Impl_t _Impl{};
+        if constexpr (_Large_function_engaged<_Vt>) {
+            _Impl._Invoke = _Function_inv_large<_Vt, _VtInvQuals, _Rx, _Noexcept, _Types...>;
+            _Impl._Move   = nullptr;
+
+            if constexpr (is_trivially_destructible_v<_Vt>) {
+                if constexpr (alignof(_Vt) <= __STDCPP_DEFAULT_NEW_ALIGNMENT__) {
+                    _Impl._Destroy = _Function_deallocate_large_default_aligned;
+                } else {
+                    _Impl._Destroy = _Function_deallocate_large_overaligned<alignof(_Vt)>;
+                }
+            } else {
+                _Impl._Destroy = _Function_destroy_large<_Vt>;
+            }
+        } else {
+            _Impl._Invoke = _Function_inv_small<_Vt, _VtInvQuals, _Rx, _Noexcept, _Types...>;
+
+            if constexpr (is_trivially_copyable_v<_Vt> && is_trivially_destructible_v<_Vt>) {
+                if constexpr ((_Function_small_copy_size<_Vt>) > _Minimum_function_size) {
+                    _Impl._Move = _Function_move_memcpy<_Function_small_copy_size<_Vt>>;
+                } else {
+                    _Impl._Move = nullptr;
+                }
+            } else {
+                _Impl._Move = _Function_move_small<_Vt>;
+            }
+
+            if constexpr (is_trivially_destructible_v<_Vt>) {
+                _Impl._Destroy = nullptr;
+            } else {
+                _Impl._Destroy = _Function_destroy_small<_Vt>;
+            }
+        }
+        return _Impl;
+    }
+
+    template <class _Vt, class _VtInvQuals>
+    _NODISCARD static const _Impl_t* _Create_impl_ptr() noexcept {
+        static constexpr _Impl_t _Impl = _Create_impl<_Vt, _VtInvQuals>();
+        return &_Impl;
+    }
+};
+
+template <class... _Signature>
+class _Move_only_function_call {
+    static_assert((_Always_false<_Signature> || ...),
+        "std::move_only_function only accepts function types as template arguments, "
+        "with possibly const/ref/noexcept qualifiers.");
+
+    static_assert(sizeof...(_Signature) > 0,
+        "Unlike std::function, std::move_only_function does not define class template argument deduction guides.");
+};
+
+// A script to generate the specializations below.
+// Avoiding C++ preprocessor for better IDE navigation and debugging experience.
+#if 0
+#include <format>

```

Comment:
Although they're tempting, we've generally moved away from macros stamping out runtime code - as @AlexGuteniev noted, they interfere with debugging, among other things. (There are still some occurrences of macros for metaprogramming, notably the horrible architecture-dependent spam for calling conventions, and some of the variant visitor logic - those are less bad because they don't directly emit code that can be stepped into.)